File: 프로그래머스/2/42747. H－Index/H－Index.java
import java.util.*;

class Solution {
    public int solution(int[] citations) {
       
        Arrays.sort(citations);
        
        int max = citations[citations.length-1];
        int cnt = 0;
        
        for(int i=citations.length-1; ;){
            if(cnt >= max) break;
            if(i >=0 && max <= citations[i]){
                cnt++;
                i--;
                continue;
            }
            max--;
        }
        
        return max;
    }
}

File: 프로그래머스/2/42747. H－Index/H－Index.js
function solution(citations) {
    
    citations.sort((a, b) => a-b);
    var max = citations[citations.length-1];
    var cnt = 0;
    
    for(let i=citations.length-1; ;){
        if(cnt >= max) break;
        if(citations[i] >= max){
            cnt++;
            i--;
            continue;
        }
        max--;
    }
    
    return max;
}